src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,74 @@
+package duke;
+
+import java.io.File;
+import java.io.IOException;
+
+import duke.commands.Command;
+import duke.exceptions.EmptyCommandException;
+import duke.exceptions.InvalidCmdValueException;
+import duke.exceptions.InvalidDateException;
+import duke.exceptions.InvalidTaskTypeException;
+import duke.exceptions.InvalidTimeException;
+import duke.tasks.TaskList;
+import duke.ui.Ui;
+import duke.utils.Parser;
+import duke.utils.Storage;
+import javafx.application.Application;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Scene;
+import javafx.scene.layout.AnchorPane;
+import javafx.stage.Stage;
+
+
+/**
+ * Duke is the class that represents the chat-bot.
+ */
+public class Duke extends Application {
+    private static final File savedFile = new File("savedFile.txt");
+    private TaskList commandList;
+    private Ui ui;
+    private Storage storage;
+    private Parser parser;
+
+    @Override
+    public void start(Stage stage) throws Exception {
```

Comment:
Perhaps you can have a Main class to control Duke and the MainWindow?

Reply:
I like it combined.